### ISSUE TITLE
Support  inequality  and equality for string types

### ISSUE THREAD

**Issue by Laschoking** [knowsys/nemo#354]
**Working:**
- Inequality of two string variables
```
foo1("hello1"). foo2("hello2"). 
literalInBody(?a,?b):-
    foo1(?a),
    foo2(?b),
    ?a != ?b.
```
- Equality of one string variable with one string literal
```
foo1("hello1").
literalInBody(?a):-
    foo1(?a),
    ?a = "hello".
```

**Not working:**
- Inequality of a string variable with one string literal

```
foo1("hello1").
literalInBody(?a):-
    foo1(?a),
    ?a != "hello".
``` 
- Equality of two string variables
```
foo1("hello1"). foo2("hello2"). 
literalInBody(?a,?b):-
    foo1(?a),
    foo2(?b),
    ?a = ?b.
```

**Comment by matzemathics** [comment]
the variable equals variable part will be fixed in #358

**Comment by aannleax** [comment]
Inequality checks are now possible between values of arbitrary types
